src/AutoRest.CSharp/Common/Output/Builders/BuilderHelpers.cs

```diff
@@ -62,9 +62,13 @@ public static Constant ParseConstant(object? value, CSharpType type)
             DateTimeSchema {Format: DateTimeSchemaFormat.DateTime} => SerializationFormat.DateTime_ISO8601,
             DateTimeSchema {Format: DateTimeSchemaFormat.DateTimeRfc1123} => SerializationFormat.DateTime_RFC1123,
             DateSchema _ => SerializationFormat.Date_ISO8601,
-            DurationSchema _ => schema.Extensions?.Format?.Equals("duration-constant") == true ? SerializationFormat.Duration_Constant : SerializationFormat.Duration_ISO8601,
+            DurationSchema _ => schema.Extensions?.Format?.Equals(XMsFormat.DurationConstant) == true ? SerializationFormat.Duration_Constant : SerializationFormat.Duration_ISO8601,
             TimeSchema _ => SerializationFormat.Time_ISO8601,
-            _ => SerializationFormat.Default
+            _ => schema.Extensions?.Format switch
+                {
+                    XMsFormat.DurationConstant => SerializationFormat.Duration_Constant,
+                    _ => SerializationFormat.Default
```

Comment:
This supports the case that only `"x-ms-format": "duration-constant"` is added to the property and we omit `"format":"duration"` which seems redundant. The schema will be treated as string type by modelerfour, therefore add a check here.